### PR TITLE
74 commoditymarketinfoview optimization

### LINKED ISSUE #74: CommodityMarketInfoView Optimization

### PR DESCRIPTION
**Pull request by pveeckhout**
closes #74

### CODE REVIEW

File: exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisSystemRepository.java
Comment (Daniel-J-Mason) [inline exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisSystemRepository.java:1]: Worth considering fully dropping coordinates.

We can create some debugging views which extract x, y, z from geom so we can view them internally as needed but not referenced by the api.

File: trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java [hunk 1]
```diff
@@ -95,23 +98,26 @@ public LocateCommodityService locateCommodityService(
 
     @Bean(name = "tradeRecieveCommodityMessageUsecase")
     public ReceiveCommodityMessageService receiveCommodityMessageService(
-            ExistsByStationNameAndSystemNameAndTimestampPort existsByStationNameAndSystemNameAndTimestamp,
-            LoadOrCreateSystemByNamePort loadOrCreateSystemByNamePort,
-            LoadOrCreateBySystemAndStationNamePort loadOrCreateBySystemAndStationNamePort,
-            LoadOrCreateCommodityByNamePort loadOrCreateCommodityByNamePort,
+            @Qualifier("tradeIdGenerator") IdGenerator idGenerator,
+            CreateOrLoadSystemPort loadOrCreateSystemByNamePort,
```
Comment (Daniel-J-Mason) [inline trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java]: rename variable to match class

File: trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java [hunk 2]
```diff
@@ -95,23 +98,26 @@ public LocateCommodityService locateCommodityService(
...
+            CreateOrLoadStationPort loadOrCreateBySystemAndStationNamePort,
```
Comment (Daniel-J-Mason) [inline trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java]: rename variable to match class

File: trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java [hunk 3]
```diff
@@ -95,23 +98,26 @@ public LocateCommodityService locateCommodityService(
...
+            CreateOrLoadCommodityPort loadOrCreateCommodityByNamePort,
```
Comment (Daniel-J-Mason) [inline trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java]: rename variable to match class

File: trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisCommodityRepository.java [hunk 4]
```diff
@@ -35,9 +33,13 @@ public interface MybatisCommodityRepository {
...
 
-    @Update("UPDATE commodity SET name = #{name} WHERE id = #{id}")
-    void update(MybatisCommodityEntity commodity);
-
-    @Delete("DELETE FROM commodity WHERE id = #{id}")
-    void delete(@Param("id") UUID id);
+    @Select({
+            "INSERT INTO commodity (id, name)",
+            "VALUES (#{id}, #{name})",
+            "ON CONFLICT name",
```
Comment (Daniel-J-Mason) [inline trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisCommodityRepository.java]: add parenthesis:

ON CONFLICT (name)

File: trade-module/src/main/resources/db/trademodule/changelog/drop_validated_market_datum_view.xml
Comment (Daniel-J-Mason) [inline trade-module/src/main/resources/db/trademodule/changelog/drop_validated_market_datum_view.xml:1]: Have we defined our inflection point yet?

Previously discussed keeping liquibase trimmed out until we decided to enter a permanent rolling db development phase

File: trade-module/src/main/java/io/edpn/backend/trade/application/port/outgoing/marketdatum/CreateWhenNotExistsLatestMarketDatumPort.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package io.edpn.backend.trade.application.port.outgoing.marketdatum;
+
+import io.edpn.backend.trade.application.domain.MarketDatum;
+
+import java.util.UUID;
+
+public interface CreateWhenNotExistsLatestMarketDatumPort {
+
+    void createWhenNotExists(UUID stationId, MarketDatum marketDatum);
```
Comment (Daniel-J-Mason) [inline trade-module/src/main/java/io/edpn/backend/trade/application/port/outgoing/marketdatum/CreateWhenNotExistsLatestMarketDatumPort.java]: Rename port to something like:
createOrUpdateExisting

This differs fundamentally from how the full MarketDatum table operates and this method should indicate that.

**Review by Daniel-J-Mason — CHANGES_REQUESTED** [review]
Additionally need to add a unique key to commodity name table to allow ON CONFLICT (name) to trigger correctly.

File: trade-module/src/main/resources/db/trademodule/changelog/drop_validated_market_datum_view.xml
Comment (pveeckhout) [reply trade-module/src/main/resources/db/trademodule/changelog/drop_validated_market_datum_view.xml:1]: generally I find it good to practise 
at this point I won't chastise anyone for drop and recreate, but ... we will need to do rolling updates in the future, might as well get used to the flow

File: exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisSystemRepository.java
Comment (pveeckhout) [reply exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisSystemRepository.java:1]: done

**Review by Daniel-J-Mason — APPROVED** [review]

**Pull request merged**